crates/sui-framework/packages/deepbook/sources/clob_v3.move

```diff
@@ -0,0 +1,3043 @@
+        // other pool info
+        base_custodian: Custodian<BaseAsset>,
+        quote_custodian: Custodian<QuoteAsset>,
+        // Stores the fee paid to create this pool. These funds are not accessible.
+        creation_fee: Balance<SUI>,
+        // Deprecated.
+        base_asset_trading_fees: Balance<BaseAsset>,
+        // Stores the trading fees paid in `QuoteAsset`. These funds are not accessible.
+        quote_asset_trading_fees: Balance<QuoteAsset>,
+    }
+
+    /// Capability granting permission to access an entry in `Pool.quote_asset_trading_fees`.
+    /// The pool objects created for older pools do not have a PoolOwnerCap because they were created
+    /// prior to the addition of this feature. Here is a list of 11 pools on mainnet that 
+    /// do not have this capability: 
+    /// 0x31d1790e617eef7f516555124155b28d663e5c600317c769a75ee6336a54c07f
+    /// 0x6e417ee1c12ad5f2600a66bc80c7bd52ff3cb7c072d508700d17cf1325324527
+    /// 0x17625f1a241d34d2da0dc113086f67a2b832e3e8cd8006887c195cd24d3598a3
+    /// 0x276ff4d99ecb3175091ba4baffa9b07590f84e2344e3f16e95d30d2c1678b84c
+    /// 0xd1f0a9baacc1864ab19534e2d4c5d6c14f2e071a1f075e8e7f9d51f2c17dc238
+    /// 0x4405b50d791fd3346754e8171aaab6bc2ed26c2c46efdd033c14b30ae507ac33
+    /// 0xf0f663cf87f1eb124da2fc9be813e0ce262146f3df60bc2052d738eb41a25899
+    /// 0xd9e45ab5440d61cc52e3b2bd915cdd643146f7593d587c715bc7bfa48311d826
+    /// 0x5deafda22b6b86127ea4299503362638bea0ca33bb212ea3a67b029356b8b955
+    /// 0x7f526b1263c4b91b43c9e646419b5696f424de28dda3c1e6658cc0a54558baa7
+    /// 0x18d871e3c3da99046dfc0d3de612c5d88859bc03b8f0568bd127d0e70dbc58be
+    struct PoolOwnerCap has key, store {
+        id: UID,
+        /// The owner of this AccountCap. Note: this is
+        /// derived from an object ID, not a user address
+        owner: address
+    }
+
+    /// Accessor functions
+    public fun usr_open_orders_exist<BaseAsset, QuoteAsset>(
+        pool: &Pool<BaseAsset, QuoteAsset>, 
+        owner: address
+    ): bool {
+        table::contains(&pool.usr_open_orders, owner)
+    }
+
+    public fun usr_open_orders_for_address<BaseAsset, QuoteAsset>(
+        pool: &Pool<BaseAsset, QuoteAsset>, 
+        owner: address
+    ): &LinkedTable<u128, u64> {
+        table::borrow(&pool.usr_open_orders, owner)
+    }
+
+    public fun usr_open_orders<BaseAsset, QuoteAsset>(
+        pool: &Pool<BaseAsset, QuoteAsset>, 
+    ): &Table<address, LinkedTable<u128, u64>> {
+        &pool.usr_open_orders
+    }
+
+    fun get_order_queue_mut(
+        tick_level: &mut TickLevel
+    ): &mut BigQueue<Order> {
+        &mut tick_level.order_queue
+    }
+
+    /// Function that takes in an order_id and then gets the tick level and the order_id of the queue
+    fun get_tick_level_and_base_order_id(
+        order_id: u128
+    ): (u64, u64) {
+        ((order_id>>64 as u64), (order_id as u64))
+    }
+
+    /// Check if the order exists in the queue
+    fun contains_order(
```

Comment:
Does this function also need to check whether the order has already been tombstoned?